ogre2/src/Ogre2DepthCamera.cc

```diff
@@ -568,7 +628,45 @@ void Ogre2DepthCamera::CreateDepthTexture()
     colorTexDef->preferDepthTexture = true;
     colorTexDef->fsaaExplicitResolve = false;
 
-    baseNodeDef->setNumTargetPass(2);
+    Ogre::TextureDefinitionBase::TextureDefinition *particleTexDef =
+        baseNodeDef->addTextureDefinition("particleTexture");
+    particleTexDef->textureType = Ogre::TEX_TYPE_2D;
+    particleTexDef->width = 0;
+    particleTexDef->height = 0;
+    particleTexDef->depth = 1;
+    particleTexDef->numMipmaps = 0;
+    particleTexDef->widthFactor = 0.5;
+    particleTexDef->heightFactor = 0.5;
+    particleTexDef->formatList = {Ogre::PF_L8};
+    particleTexDef->fsaa = 0;
+    particleTexDef->uav = false;
+    particleTexDef->automipmaps = false;
+    particleTexDef->hwGammaWrite = Ogre::TextureDefinitionBase::BoolFalse;
+    particleTexDef->depthBufferId = Ogre::DepthBuffer::POOL_DEFAULT;
+    particleTexDef->depthBufferFormat = Ogre::PF_UNKNOWN;
+    particleTexDef->preferDepthTexture = false;
+    particleTexDef->fsaaExplicitResolve = false;
+
+    Ogre::TextureDefinitionBase::TextureDefinition *particleDepthTexDef =
+        baseNodeDef->addTextureDefinition("particleDepthTexture");
+    particleDepthTexDef->textureType = Ogre::TEX_TYPE_2D;
+    particleDepthTexDef->width = 0;
+    particleDepthTexDef->height = 0;
+    particleDepthTexDef->depth = 1;
+    particleDepthTexDef->numMipmaps = 0;
+    particleDepthTexDef->widthFactor = 0.5;
+    particleDepthTexDef->heightFactor = 0.5;
+    particleDepthTexDef->formatList = {Ogre::PF_D32_FLOAT};
+    particleDepthTexDef->fsaa = 0;
+    particleDepthTexDef->uav = false;
+    particleDepthTexDef->automipmaps = false;
+    particleDepthTexDef->hwGammaWrite = Ogre::TextureDefinitionBase::BoolFalse;
+    particleDepthTexDef->depthBufferId = Ogre::DepthBuffer::POOL_NON_SHAREABLE;
+    particleDepthTexDef->depthBufferFormat = Ogre::PF_UNKNOWN;
+    particleDepthTexDef->fsaaExplicitResolve = false;
+
+    baseNodeDef->setNumTargetPass(5);
```

Comment:
it went from:

```
target colorTexture
target rt0
```

to

```
target colorTexture
target depthTexture
target particleTexture
target particleDepthTexture
target rt0
```

The extra passes are used as masks for the particles in order to segment them out from the image so that I can apply the scattering effect to only the particle pixels

Reply:
I see this now, I was confusing the number of target passes (5) with the number of texture definitions (6). Thanks :+1: